renku/api/datasets.py

```diff
@@ -449,6 +451,65 @@ def get_relative_url(self, url):
             url = str(url / submodule_url.name)
         return url
 
+    def get_newest_commit_for_dataset(self, dataset):
```

Comment:
To make this useful in the future (and possible to track lineage). We should probably build an ordered index which contains all commits in which given dataset was referenced. Ensure that commits are ordered and then simply query for head or tail.